client/me/purchases/membership-item/index.tsx

```diff
@@ -79,7 +81,7 @@ const MembershipType = ( { subscription }: { subscription: MembershipSubscriptio
 
 const Icon = ( { subscription }: { subscription: MembershipSubscription } ) => {
 	const [ hasError, setErrors ] = useState( false );
-	const [ site, setSite ] = useState( null );
+	const [ site, setSite ] = useState< null | { icon?: { ico: string } } >( null );
```

Comment:
if `site` doesn't need to be `null` and could be `undefined` instead, then I believe that `setState` already acknowledges this.

```ts
const [ site, setSite ] = useState< { icon?: { ico: string } } >();
// ?^ site is { icon?: { ico: string } } | undefined
```